app/marketing/mails.py

```diff
@@ -300,18 +301,24 @@ def new_supporter(grant, subscription):
         translation.activate(cur_language)
 
 
-def thank_you_for_supporting(grant, subscription):
-    if subscription and subscription.negative:
-        return
+def thank_you_for_supporting(grants, contributor):
+    subscriptions = []
+    for (grant_id, payload) in grants:
+        subscription = Subscription.objects.filter(grant_id=grant_id, contributor_profile=contributor).first()
```

Comment:
If i made a contribution to grant A, grant B -> I'd get a mail for it
Now a while later if i make another contribution for grant B + grant C

Wouldn't my grant A contribution also turn up in the mail?

Reply:
My main motivation to use this approach was because it received a prior approval; but I don't think so, it would only grab the subscriptions from the grants in the most recent bulk cart payload, they would however receive another email with grant B in the later contribution.

Maybe this addresses your concern: https://github.com/gitcoinco/web/pull/7880/commits/04c423222b8d9cf3cf9b605ac83364099fab6974 

If we went in this direction, it would only send emails after the grant processing task successfully completes.